napari/layers/shapes/_shape_list.py

```diff
@@ -69,11 +69,11 @@ class ShapeList:
 
     def __init__(self, data=(), ndisplay=2) -> None:
         self._ndisplay = ndisplay
-        self.shapes = []
-        self._displayed = []
-        self._slice_key = []
```

Comment:
This is triggering a failure in test. 

This is dues to numpy's `==` that have deprecated comparison of differently sized arrays. I belive you now need to use `np.array_equal`

Reply:
```
diff --git a/napari/layers/shapes/_shape_list.py b/napari/layers/shapes/_shape_list.py
index 3677b5a15..11861d4f7 100644
--- a/napari/layers/shapes/_shape_list.py
+++ b/napari/layers/shapes/_shape_list.py
@@ -187,7 +187,7 @@ class ShapeList:
     @slice_key.setter
     def slice_key(self, slice_key):
         slice_key = list(slice_key)
-        if not np.all(self._slice_key == slice_key):
+        if not np.array_equal(self._slice_key, slice_key):
             self._slice_key = slice_key
             self._update_displayed()
```
Seem to fix most of the failure, but seem to actually create some genuine issues down the road, which might be hidden bugs, or broadcast issues.

Reply:
Ok, there is a difference in broadcasting: 

```
In [5]: np.array_equal(np.array([[1,2],[1,2]]),  np.array([1,2]))
Out[5]: False

In [6]: np.all(np.array([[1,2],[1,2]]) == np.array([1,2]))
Out[6]: True
```